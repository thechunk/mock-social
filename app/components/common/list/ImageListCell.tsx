import React, {PureComponent} from 'react';
import {Text, View} from 'react-native';
import ListCell from './ListCell';
import WrappedWithPhotos from "../album-container/WrappedWithPhotos";
import styles, {inherited} from "../../album-list/styles";
import LoadedImage from "../image/LoadedImage";

class ImageListCell extends PureComponent<IImageListCell> {
    render() {
        return (
            <ListCell
                onPress={this.props.onPress}
                containerStyle={inherited.cellOuter}
                cellStyle={this.props.cellStyle}>
                <WrappedWithPhotos
                    id={this.props.id}
                    renderItem={v => <LoadedImage style={this.props.imageStyle} source={{uri: v.url}} />} />
                <View style={styles.cellTextView}>
                    <Text style={this.props.titleStyle}>{this.props.title}</Text>
                    {this.props.subtitle && this.props.subtitle.length > 0
                        ? <Text style={this.props.subtitleStyle}>{this.props.subtitle}</Text>
                        : null
                    }
                </View>
            </ListCell>
        );
    }
}

export default ImageListCell;