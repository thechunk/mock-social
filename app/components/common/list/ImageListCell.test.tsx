import * as React from 'react';
import {shallow} from "enzyme";
import ImageListCell from "./ImageListCell";

describe('ImageListCell', () => {
    it('renders correctly', () => {
        expect(shallow(<ImageListCell id={1} key={1} title="title"/>)).toMatchSnapshot();
    });
});