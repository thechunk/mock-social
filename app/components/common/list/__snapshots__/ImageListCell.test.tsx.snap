// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageListCell renders correctly 1`] = `
<ListCell
  containerStyle={
    Object {
      "borderBottomColor": "rgba(0, 0, 0, 0.2)",
      "borderBottomWidth": 0.5,
    }
  }
>
  <Component
    id={1}
    renderItem={[Function]}
  />
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "marginLeft": 8,
      }
    }
  >
    <Text
      numberOfLines={2}
    >
      title
    </Text>
  </View>
</ListCell>
`;
